### DISCUSSION TITLE
How is this different ?

### DISCUSSION THREAD

**Discussion by kepta** [prosekit/prosekit#857] [General]
Hello, I have been a long time lurker of PM ecosystem and particularly of your work in remirror and have nothing but admiration for you.

Since you started this new project I wanted to ask how is it different from Remirror, Prosemirror or Tiptap?

Thanks again !

**Comment by ocavue** [comment]
Thanks for your kind words.

All three other editors mentioned are based on ProseMirror. ProseMirror is the solid foundation.

Remirror and TipTap share some similar designs. One of them is that every extension is a class (e.g., `BoldExtension`). If you want to add more features based on an existing extension, you extend it (e.g., `class MyBoldExtension extends BoldExtension`) and add/override methods in the subclass.

I found this design makes it harder to use in complex scenarios. Using the principle of "composition over inheritance," I redesigned the extension system in ProseKit, where every extension is just a function that calls one or more other extensions.

I also made some other improvements, like simpler lifecycle management and more UI library (Svelte, SolidJS, etc.) support.

Sadly, these changes are so significant that I won't be able to publish them as a new major version of the existing Remirror project, so I created ProseKit.

**Reply by mustafa0x** [reply]
prosekit is quite impressive, thanks @ocavue. is there a migration guide? any general advice? do all/most extensions exist in prosekit? (that seems to be the case from what i could tell)
